### ISSUE TITLE
Worker volume transfer speed is very slow (and has been since 2.7.1)

### ISSUE THREAD

**Issue by baszalmstra** [concourse/concourse#1030]
Volume transfers from a linux worker to a windows worker seem to slow down considerably since 2.7.1. With version 2.7.0 I easily reached 950Mbit/s transfer rate on a 1Gb network. Since upgrading concourse to version 2.7.1 its down to 3.5Mbit/s. 

My case:
Transferring a single git resource that was fetched on a linux worker and required on a windows worker. Something along the lines of:
```
- get: source
- task: build
  file: ...
```
The server running the linux worker is also running the atc. I'm simply starting both with `concourse web` and `concourse worker`. Everything was fine until version 2.7.1. 

I tried the following to see if it was concourse or something else:

- disabled all other processes on both machines
- ran transfer tests (with iperf) while transferring to see if it's a resource issue but came out with nothing (the transfer tests transferred at 950Mbit/s)
- the worker and web instance of concourse only use about 1 maybe 2% of CPU and little memory.
- tested with multiple windows workers
- running concourse workers as admin on windows.
- there are no errors in logs

I noticed the issue is introduced in 2.7.1 by clean installing version 2.7.0 through version 2.7.3. Only with 2.7.0 the data is transferred at full speed.

Also another symptom is that if I just wait (a few hours in my case) the transfer fails with `tar extract failed (exit status 1).` when creating a symlink (on windows). This doesn't happen with 2.7.0.

**Comment by joshzarrabi** [comment]
@baszalmstra This is interesting, because we fixed a TSA memory leak on 2.7.1 which I would assume would have the opposite effect of this. You can see this commit here: https://github.com/concourse/tsa/compare/000416a7c6b610be8997d8f3893eda7fdef7d52f...887e63f407654611e3c7bb15c474699027619bcc#diff-22362baea3687168b69d16fac9125ea6. 

Can we have more information about your deployment... What IAAS are you running on, do you have a special network set up in any way?

**Comment by baszalmstra** [comment]
@joshzarrabi I'm running concourse on 4 servers using the binaries and a dedicated database server (running with some other database and a private docker registry). We have 3 windows machines and 1 linux machine. The linux machine is solely running `concourse web` as well as a single worker. The windows machines are only running `concourse worker`. All machines are connected to a single 1Gb switch which is also connected to our office network. The linux machine is running a fresh ubuntu 16.04.2 LTS install. The windows machines are running Windows Server 2012 R2. I have not touched any of the network configurations except for trying to disable the firewalls on the windows machines (which didn't help). 

I have to mention that I have seen this problem before when I didn't specify the `--peer-ip` on the workers when I set the whole thing up initially. That also resulted in slow transfer rates although I can't be sure if it resulted in the same transfer rates I have with 2.7.1 and up.

I have no experience with go but I'm very happy to help out anyway I can. I can quite easily install the entire stack from scratch so upgrading and downgrading versions is no problem.

**Comment by joshzarrabi** [comment]
@baszalmstra Cool, thanks for the detailed information. If this is really biting you right now I would downgrade, as we have not really heard this feedback from many other people and don't really have a plan of action. We'll keep this on our radar though.

**Comment by baszalmstra** [comment]
Thats too bad. Ill stick with 2.7.0 for now.

**Comment by clarafu** [comment]
Scheduling this soonish so we can at least take a look, since it sounds like you've nailed it down to a version (thanks!).

**Comment by baszalmstra** [comment]
Any progress on this issue? Anything I can do to help? I would like to upgrade to 3.x but cant because of this issue.

**Comment by mariash** [comment]
@baszalmstra do you see this issue in 3.1.1?

**Comment by baszalmstra** [comment]
I just upgraded everything from 2.7.0 to 3.1.1 and I can confirm that this is still happening with 3.1.1. I'll stick with 2.7.0 for now because builds are waaayyy to slow with 3.1.1. 

Please let me know if I can provide you with logs/screenshots/configurations/etc if that would help

**Comment by baszalmstra** [comment]
Has there been any progress on this issue? It really bums me out that I can't upgrade. If I can help out in any way with solving this issue do let met know!

**Comment by baszalmstra** [comment]
Changed all network hardware, the issue still persists.
Upgrading from 2.7.0 to 3.8.0, the issue still persists. 
Upgraded from Windows Server 2012R2 to Windows Server 2016, issue still persists.

I've done a lot of performance monitoring and all I have been able to find is that the transfer speed between the atc and the worker is just slow. I did notice that all transfer traffic is always routed through the atc (even though I specified --peer-ip). 

The Send-Q in `ss` also shows a very high value (400.000+) for the connection between the atc and the worker when transferring. I could not really find a cause for this.

Any help is much appreciated.

**Comment by baszalmstra** [comment]
I stumbled upon something interesting. I accidentally forgot to downgrade one of my windows workers from 3.8.0 to 2.7.0. I did downgrade the ATC. I noticed that this results in the same behavior which leads me to expect that the problem lies somewhere on the windows side of things.

Again, any help is much appreciated. I would also be very happy to provide server/process/whatever information, I just have no idea what is helpful..

**Comment by baszalmstra** [comment]
It appears I've solved this issue while moving from a chef based installation approach to ansible. I can't be sure what resolved the issue. I've removed all chef components from the servers, disabled Real-Time protection and poof since today my transfer speeds between windows and linux servers went from 4Mbit to 900Mbit.

Very strange, but good riddance, now I can finally keep up with the latest version again, especially looking forward to spaces!
